src/coreComponents/fileIO/timeHistory/PackCollection.cpp

```diff
@@ -57,99 +56,140 @@ void PackCollection::initializePostSubGroups( )
   }
 }
 
-HistoryMetadata PackCollection::getMetadata( DomainPartition const & domain, localIndex collectionIdx )
+HistoryMetadata PackCollection::getMetaData( DomainPartition const & domain, localIndex collectionIdx ) const
 {
   Group const * targetObject = this->getTargetObject( domain, m_objectPath );
-  WrapperBase const & target = targetObject->getWrapperBase( m_fieldName );
-  if( m_setNames.size() != 0 )
+  WrapperBase const & targetField = targetObject->getWrapperBase( m_fieldName );
+
+  if( collectAll() )
+  {
+    localIndex const packCount = m_setsIndices[0].size() == 0 ? targetField.size() : m_setsIndices[0].size(); // TODO CHECK!
```

Comment:
This looks OK; I've been in a case where I could test it.
Add comments about what `m_setsIndices` means and why we take this first index.

Reply:
done